brave/src/main/java/brave/propagation/StrictScopeDecorator.java

```diff
@@ -40,13 +40,11 @@
  * }</pre>
  */
 // Closeable so things like Spring will automatically execute it on shutdown and expose leaks!
-public final class StrictScopeDecorator implements ScopeDecorator, Closeable {
+public final class StrictScopeDecorator extends WeakConcurrentMap<Scope, CallerStackTrace> implements ScopeDecorator, Closeable {
```

Comment:
I might suggest having the WeakConcurrentMap type be an inner class vs leaking it in the signature

If you have more time, bonus points to see if upstream works now (including shading it). I think raphael changed it to allow our features to work.

Reply:
what I meant here was a package-private inner static class like `ScopeToCallerStackTrace extends WeakConcurrentMap`. That would allow you to call methods you wanted exposed. It is another ref, but StrictScopeDecorator is a singleton, so shouldn't matter. The class will be slightly larger, that's it!

Reply:
Yup good call, changed to an inner class. Looks like the changes for the upstream library haven't been released yet so after that we can try cleaning up.